### ISSUE TITLE
Various prop types are missing

### ISSUE THREAD

**Issue by jaredly** [react-community/react-navigation#4]
From the code, there are a bunch of places where prop types are used, but not indicated in other places.
- screenProps -> apparently these can be passed down, but there's little indication
- making a custom header; StackNavigator passes in `HeaderComponent` to `CardStack` but CardStack doesn't seem to do anything with it

**Comment by ericvicenti** [comment]
Hey Jared!

As for CardStack, yeah there are a handful of old props left over from the NavigationExperimental views that were moved over. For now I think the best approach is to clean them up, and we can revisit adding things in later.

@satya164, could you make sure the flow types and docs are up to date for `screenProps`?

**Comment by satya164** [comment]
@ericvicenti `screenProps` are typed as `{}` since they can hold arbitrary objects. Not sure what I have to change there.

**Comment by coodoo** [comment]
As a side note, while reading the docs I got a bit confused by whether `screenProps` is the same thing as `this.props.navigation` object in each screen, maybe it was just an old name?

**Comment by ericvicenti** [comment]
@coodoo, `screenProps` are extra props that will be passed through the navigator into each screen component, in *addition* to the navigation prop. Could you submit a PR to help clarify the docs?

**Comment by jaredly** [comment]
@satya164 `screenProps` are not listed in a bunch of places -- for example, I believe that [`NavigationContainer`](https://github.com/react-community/react-navigation/blob/master/src/createNavigationContainer.js#L28) should list `screenProps` as a thing it accepts

**Comment by satya164** [comment]
@jaredly `NavigationContainer` doesn't accept `screenProps` - https://github.com/react-community/react-navigation/search?utf8=%E2%9C%93&q=screenProps&type=Code

**Comment by jaredly** [comment]
@satya164 I believe that it does

Here's the trail:

from [`StackNavigator`](https://github.com/react-community/react-navigation/blob/master/src/navigators/StackNavigator.js#L41)
```js
  return createNavigationContainer(createNavigator(router)(props => (
    <CardStack
      {...props}
      headerComponent={headerComponent}
      headerMode={headerMode}
      mode={mode}
    />
  )), containerOptions);
```
[`createNavigationContainer`](https://github.com/react-community/react-navigation/blob/master/src/createNavigationContainer.js#L157) and [`createNavigator`](https://github.com/react-community/react-navigation/blob/master/src/navigators/createNavigator.js#L24) both spread in all props. 

And then [`CardStack`](https://github.com/react-community/react-navigation/blob/master/src/views/CardStack.js#L43) accepts `screenProps`.

TLDR I had to read a ton of source code to realize that I could do
```
const SomeStack = StackNavigator({
  // config
});

<SomeStack
  screenProps={/* these will get passed to the screen components */}
/>
```

Please fix the docs so future users don't have to read all your source code.

**Comment by coodoo** [comment]
@ericvicenti Absolutely, will see how to merge all above mentioned details into the docs.

**Comment by grabbou** [comment]
Closing as we are now generating prop-types directly from `flow` which should make it more up-to-date.
